spark/src/main/java/org/apache/iceberg/actions/RemoveOrphanFilesAction.java

```diff
@@ -79,6 +87,20 @@
     }
   }, DataTypes.StringType);
 
+  private static final UserDefinedFunction decode = functions.udf((String fullyQualifiedPath) -> {
+    return URLDecoder.decode(fullyQualifiedPath, "UTF-8");
+  }, DataTypes.StringType);
+
+  /**
+   * transform a file path to
+   * {@code Dataset<Row<file_path_no_scheme_authority, file_path_with_scheme_authority>>}
+   */
+  private static final UserDefinedFunction fileWithRelativePath = functions.udf((String fullyQualifiedPath) -> {
```

Comment:
I don't want to bike shed to much here, but I think this name ends up being confusing because later we use it in conjunction with very similarly named variables which represent Dataframes and also use Apply methods. I would recommend we rename this to something with "UDF" in it so it's clear.